src/Microsoft.Azure.NotificationHubs/Messaging/MessagingUtilities.cs

```diff
@@ -0,0 +1,83 @@
+﻿//------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved. 
+// Licensed under the MIT License. See License.txt in the project root for 
+// license information.
+//------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.NotificationHubs.Messaging
+{
+    public static class MessagingUtilities
+    {
+        public static void ThrowIfNullAddressOrPathExists(Uri address)
+        {
+            if (address == null)
+            {
+                throw new ArgumentNullException(nameof(address));
+            }
+
+            if (!string.IsNullOrEmpty(address.AbsolutePath) && address.Segments.Length > 3)
+            {
+                throw new ArgumentException(SRClient.InvalidAddressPath(address.AbsoluteUri), nameof(address));
+            }
+        }
+
+        /// <summary>
+        /// Create a list of uri addresses from a given list of string addresses
+        /// </summary>
+        /// <param name="addresses"></param>
+        /// <returns></returns>
+        public static IEnumerable<Uri> GetUriList(IEnumerable<string> addresses)
```

Comment:
Is there a way to use IList instead because of the possibility of multiple enumeration?